src/EventStore.Client/Common/Diagnostics/EventMetadataExtensions.cs

```diff
@@ -17,18 +17,14 @@ static class EventMetadataExtensions {
 
 	[MethodImpl(MethodImplOptions.AggressiveInlining)]
 	public static TracingMetadata ExtractTracingMetadata(this ReadOnlyMemory<byte> eventMetadata) {
-		try {
-			using var doc = JsonDocument.Parse(eventMetadata);
-
-			if (!doc.RootElement.TryGetProperty(TracingConstants.Metadata.TraceId, out var traceId)
-			 || !doc.RootElement.TryGetProperty(TracingConstants.Metadata.SpanId, out var spanId))
-				return TracingMetadata.None;
-
-			return new TracingMetadata(traceId.GetString(), spanId.GetString());
-		}
-		catch (JsonException) {
+		var reader = new Utf8JsonReader(eventMetadata.Span);
+		
+		if (!JsonDocument.TryParseValue(ref reader, out var doc)
```

Comment:
Unless I'm misreading, you just removed the dispose of doc (because you're no longer `using` it). This does have an impact ... https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsondocument?view=net-8.0#remarks

Reply:
Hey @yreynhout I think you're looking at an earlier commit.

https://github.com/EventStore/EventStore-Client-Dotnet/commit/12b6e39fb411ff6c824b7568576598d86e20977b#diff-a51f5117c6e94e4d8fa313ae0514855ad5245622618c0ff8d2aeb19a6d9b1b26R27

Reply:
FML, I clicked a single commit instead of viewing all file changes ... karma. All good.